lib/screens/home/explorer/explorer_bloc.dart

```diff
@@ -1,32 +1,236 @@
+
+      abbrValidators = await _validatorService.getAbbreviatedValidators(
+        account.coin,
+        1,
+      );
+
+      _abbreviatedValidators.addAll(abbrValidators);
+
+      final delegations = await _validatorService.getDelegations(
+        _accountDetails.coin,
+        _accountDetails.address,
+        _delegationPages.value,
+        _stakingDetails.value.selectedState,
+      );
+
+      final validators = await _validatorService.getRecentValidators(
+        _accountDetails.coin,
+        _validatorPages.value,
+        _stakingDetails.value.selectedStatus,
+      );
+
+      _stakingDetails.tryAdd(
+        StakingDetails(
+          abbreviatedValidators: _abbreviatedValidators,
+          delegates: delegations,
+          validators: validators,
+          address: account.address,
+        ),
+      );
+    } finally {
+      if (showLoading) {
+        _isLoading.tryAdd(false);
+      }
+    }
+  }
+
+  Future<void> updateState(DelegationState state) async {
+    final oldDetails = _stakingDetails.value;
+
+    if (state == oldDetails.selectedState) {
+      return;
+    }
+
+    _delegationPages.value = 1;
+
+    final delegations = await _validatorService.getDelegations(
+        _accountDetails.coin,
+        _accountDetails.address,
+        _delegationPages.value,
+        state);
+
+    _stakingDetails.tryAdd(
+      StakingDetails(
+        abbreviatedValidators: _abbreviatedValidators,
+        delegates: delegations,
+        validators: oldDetails.validators,
+        address: oldDetails.address,
+        selectedState: state,
+        selectedStatus: oldDetails.selectedStatus,
+      ),
+    );
+  }
+
+  Future<void> updateStatus(ValidatorStatus status) async {
+    final oldDetails = _stakingDetails.value;
+
+    if (status == oldDetails.selectedStatus) {
+      return;
+    }
+
+    _validatorPages.value = 1;
+
+    final validators = await _validatorService.getRecentValidators(
+      _accountDetails.coin,
+      _validatorPages.value,
+      status,
+    );
+
+    _stakingDetails.tryAdd(
+      StakingDetails(
+        abbreviatedValidators: _abbreviatedValidators,
+        delegates: oldDetails.delegates,
+        validators: validators,
+        address: oldDetails.address,
+        selectedState: oldDetails.selectedState,
+        selectedStatus: status,
+      ),
+    );
+  }
+
+  Future<List<T>> _loadMore<T>(
+      List<T> oldList,
+      BehaviorSubject<int> pages,
+      BehaviorSubject<bool> isLoading,
+      Future<List<T>> Function() function) async {
+    if (pages.value * 30 > oldList.length) {
+      return oldList;
+    }
+    pages.value++;
+    isLoading.value = true;
+    final newList = await function();
+
+    if (newList.isNotEmpty) {
+      oldList.addAll(newList);
+    }
+    return oldList;
+  }
+
+  Future<void> loadAdditionalDelegates() async {
+    var oldDetails = _stakingDetails.value;
+
+    final delegates = await _loadMore(
```

Comment:
Might be worth creating a generic paged cache class that can work for this and other paged services, and write unit tests around it.

Reply:
It will take a bit of brainstorming, but I love it.